### PR TITLE
refactor(blockstore): footprint and blockstore wksp obj creation

### PR DESCRIPTION
**Pull request by lidatong**

### CODE REVIEW

File: src/flamenco/runtime/fd_blockstore.c [hunk 1]
```diff
@@ -33,59 +33,78 @@ fd_blockstore_align( void ) {
...
   }
 
   fd_wksp_t * wksp = fd_wksp_containing( blockstore );
   if( FD_UNLIKELY( !wksp ) ) {
-    FD_LOG_WARNING( ( "shmem must be part of a workspace" ) );
+    FD_LOG_WARNING(( "shmem must be part of a workspace" ));
     return NULL;
   }
 
+  FD_LOG_NOTICE(("shred_max=%lu block_max=%lu txn_max=%lu alloc_max=%lu", shred_max, block_max, txn_max, alloc_max ));
```
Comment (asiegel-jt) [inline src/flamenco/runtime/fd_blockstore.c:88]: Remove or make prettier

File: src/flamenco/runtime/fd_blockstore.c [hunk 2]
```diff
@@ -33,59 +33,78 @@ fd_blockstore_align( void ) {
 }
 
 ulong
-fd_blockstore_footprint( void ) {
-  return sizeof( fd_blockstore_t );
+fd_blockstore_footprint( ulong shred_max, ulong block_max, ulong txn_max, ulong alloc_max ) {
+  ulong l = FD_LAYOUT_FINI(
```
Comment (asiegel-jt) [inline src/flamenco/runtime/fd_blockstore.c:37]: You are using fd_wksp_alloc_laddr to allocate the parts, which is inconsistent with just laying out the pieces consecutively

File: src/flamenco/runtime/fd_blockstore.c [hunk 3]
```diff
@@ -306,80 +185,118 @@ fd_blockstore_delete( void * shblockstore ) {
...
 
-  /* Free all blocks. */
+  /* Delete all structures. */
 
-  ulong * q = fd_wksp_laddr_fast( wksp, blockstore->slot_deque_gaddr );
-  fd_blockstore_slot_deque_remove_all( q );
-  fd_blockstore_slot_deque_push_tail( q, blockstore->smr );
-  while( !fd_blockstore_slot_deque_empty( q ) ) {
-    ulong curr = fd_blockstore_slot_deque_pop_head( q );
+  blockstore->shred_pool = fd_buf_shred_pool_delete( blockstore->shred_pool );
```
Comment (asiegel-jt) [inline src/flamenco/runtime/fd_blockstore.c:204]: why are you overwriting the pointers. you can just use (void) on the result of delete

File: src/flamenco/runtime/fd_blockstore.c [hunk 4]
```diff
@@ -306,80 +185,118 @@ fd_blockstore_delete( void * shblockstore ) {
...
+                                       FD_BLOCK_FLAG_CONFIRMED ),
+                                       FD_BLOCK_FLAG_FINALIZED );
+  block_map_entry->reference_tick  = 0;
+  block_map_entry->ts              = 0;
+
+  block_map_entry->consumed_idx    = 0;
+  block_map_entry->received_idx    = 0;
+  block_map_entry->complete_idx    = 0;
+
+  /* This creates an empty allocation for a block, to "facade" that we
```
Comment (asiegel-jt) [inline src/flamenco/runtime/fd_blockstore.c:253]: facade isn't a verb :)

**Review by asiegel-jt — APPROVED** [review]

File: src/flamenco/runtime/fd_blockstore.c (hunk 4 above)
Comment (lidatong) [reply src/flamenco/runtime/fd_blockstore.c:253]: lol

**Pull request merged**